components/NavBar/index.js

```diff
@@ -1,68 +1,242 @@
-import Link from 'next/link';
-import React, { useState, useRef } from 'react';
+import React, { useState, useRef, useEffect } from 'react';
+import { useRouter } from 'next/router';
 import styles from './navbar.module.css';
+import Link from 'next/link';
 import Image from 'next/image';
 import GenericClosePopUp from '../Close-popup/GenericClosePopUp';
+import { USER_DATA_URL, LOGIN_URL, PATHS } from 'constants.js';
 
-const NavBar = ({ personData: { photo } }) => {
-  const RDSLogo = '/assets/Real-Dev-Squad1x.png';
+const NavBar = () => {
+  const router = useRouter();
+
+  const RDS_LOGO = '/assets/Real-Dev-Squad1x.png';
+  const GITHUB_LOGO = '/assets/github.png';
+  const DEFAULT_AVATAR = '/assets/default_avatar.jpg';
+  const [userData, setUserData] = useState({});
   const [toggle, setToggle] = useState(false);
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  const [mountedComponent, setMountedComponent] = useState(false);
   const navbarRef = useRef();
   GenericClosePopUp(navbarRef, () => {
     setToggle(false);
   });
 
-  return (
-    <nav className={styles.navBar}>
-      <ul>
-        <li>
-          <div className={styles.logo}>
-            <Link href="https://realdevsquad.com">
-              <Image src={RDSLogo} alt="Navbar Logo" width="50" height="50" />
-            </Link>
-          </div>
-        </li>
-        {/* <li>
-          <Link href="/shop">
-            <a>Shop</a>
-          </Link>
-        </li>
-        <li>
-          <Link href="/cart">
-            <a>Cart</a>
-          </Link>
-        </li> */}
-        <li>
-          <Link href="/auction">
-            <a>Auction</a>
-          </Link>
-        </li>
-        <li>
-          <Link href="/trading">
-            <a>Stocks</a>
-          </Link>
-        </li>
-      </ul>
+  useEffect(() => {
+    const fetchData = async () => {
+      await fetch(USER_DATA_URL, { credentials: 'include' })
+        .then((response) => {
+          if (!response.ok) {
+            setIsLoggedIn(false);
+            throw new Error(`${response.status} (${response.statusText})`);
+          }
+          return response.json();
+        })
+        .then((responseJson) => {
+          if (responseJson.incompleteUserDetails) {
+            return window.location.replace(
+              'https://my.realdevsquad.com/signup'
+            );
+          }
```

Comment:
1. Redirecting the user without letting them know anything? The user comes to crypto site, doesnt have a cookie, gets redirected to a github login page?! Wouldn't the user be confused?
2. The redirection should be in a new tab, right? After logging in with github, we redirect the user to realdevsquad.com. That's a dead end. Crypto page is lost now.

Reply:
Will be addressed in #246